### PR TITLE
Lesson 02 finished

### PR DESCRIPTION
**Pull request by gjkim44**
I finished lesson02 assignments.  I got help from you tube for grid printer.   I also used the series_template to finish the fibonacci series exercise.

### CODE REVIEW

File: students/GKim/lesson02/series.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    so that this function works for any first two numbers for a sum series.
+    Once generalized that way, sum_series(n, 0, 1) should be equivalent to fibonacci(n).
+    And sum_series(n, 2, 1) should be equivalent to lucas(n).
+    """
+    if n0 == None and n1 == None:
+        return fibonacci(n)
+
+    elif n0 == 2 and n1 == 1:
+        return lucas(n)
+
```
Comment (nairraghav) [inline students/GKim/lesson02/series.py]: This is very close! Remember that you will also have to handle series that don't start with 2,1 or no optional parameters. If n0 = 5 and n1 = 2 (or almost any other permutation), this sum_series won't do anything.

Hint: Your fibonacci and lucas functions are extremely similar. Think about how you can use one general function to do both of their jobs

File: students/GKim/lesson02/grid_printer.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    """
+    prints the "columns" " | " in the grid. just like the row first section is 
+    already created by the input given from user.  The colums print across 
+    """
+    print("|", end = " ")
+    for _ in range(c - 1):
+        print("        |", end = " ")
+    print("        |")
+
+def my_grid(c,r):
```
Comment (nairraghav) [inline students/GKim/lesson02/grid_printer.py]: This is close but it doesn't achieve what we are aiming for. This prints out the rows/columns but we are looking for a grid where the rows/columns are the same. We are looking to grind a grid (with same rows/columns) but the **size** of each individual cell is another factor.

**Comment by nairraghav** [comment]
Series looks good! Please take a look at the Grid Printer comments :)

File: students/GKim/lesson02/grid_printer.py (hunk 2 above)
Comment (gjkim44) [reply students/GKim/lesson02/grid_printer.py]: Ok I will make changes

File: students/GKim/lesson02/series.py [hunk 3]
```diff
@@ -33,12 +33,13 @@ def sum_series(n, n0=None, n1=None):
     Once generalized that way, sum_series(n, 0, 1) should be equivalent to fibonacci(n).
     And sum_series(n, 2, 1) should be equivalent to lucas(n).
     """
-    if n0 == None and n1 == None:
-        return fibonacci(n)
-
-    elif n0 == 2 and n1 == 1:
-        return lucas(n)
-
+    if n == 0:
```
Comment (nairraghav) [inline students/GKim/lesson02/series.py:36]: Nice job generalizing the function!

File: students/GKim/lesson02/grid_printer.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    prints the "columns" " | " in the grid. just like the row first section is 
+    already created by the input given from user.  The colums print across 
+    """
+    for _ in range(a-1):
+        print("|", end = "")
+        print("   " * ((c-1)//2), end = "")
+    print("|")
+    
+
+def my_grid(c,a =3):
```
Comment (nairraghav) [inline students/GKim/lesson02/grid_printer.py]: This looks to be printing properly when given 1 parameter, ex: my_grid(5), my_grid(7). 

It doesn't properly handle the case when we pass 2 parameters though. The first parameter determines the nxn grid size and the second parameter determines the size of each cell. So for example, print_grid(5,3) should return a 5x5 grid where each cell is 3x3 (3 '-' across and 3 '|' long )

File: students/GKim/lesson02/series.py (hunk 1 above)
Comment (nairraghav) [reply students/GKim/lesson02/series.py]: This looks good, nice job!

**Comment by gjkim44** [comment]
I shortened up the grid printer file to have 3 total functions!

File: students/GKim/lesson02/fizz_buzz.py [hunk 5]
```diff
@@ -1,6 +1,4 @@
 
-
-
 for x in range(1,101):
     if x % 15 == 0:
```
Comment (PythonCHB) [inline students/GKim/lesson02/fizz_buzz.py:3]: nice catch with the % 15 :-)

File: students/GKim/lesson02/grid_printer.py [hunk 6]
```diff
@@ -0,0 +1,118 @@
+
+"""
+Shortened version
+"""
+
+def print_symbol(c, a =2,s1="",s2=""):
```
Comment (PythonCHB) [inline students/GKim/lesson02/grid_printer.py:6]: be consistent with your spacing, ideally following PEP8:

```
def print_symbol(c, a=2, s1="", s2=""):
```
make it a little more readable ...

File: students/GKim/lesson02/grid_printer.py [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    """
+    function to print + and - using a 
+    for loop .  Using input from user the range is -1 because
+    from the first intial print the  section of the column is already created
+    this all prints accross
+    """
+    if c % 2 == 0:
+        x = int(c)/2
+    elif c % 2 == 1:
+        x = int(c-1)/2
```
Comment (PythonCHB) [inline students/GKim/lesson02/grid_printer.py:16]: you can replace all this logic with "floor division":

```
x = c // 2
```

File: students/GKim/lesson02/grid_printer.py [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+    for _ in range(a):
+        print("{}".format(s1), end = "")
```
Comment (PythonCHB) [inline students/GKim/lesson02/grid_printer.py:18]: This: `"{}".format(s1)` is EXACTLY the same as simply `s1` :-)

File: students/GKim/lesson02/series.py [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+    return b
+    
+
+
+def lucas(n):
+    """ compute the nth Lucas number """
+    a, b = 2, 1
+    if n == 0:
+        return a
+    elif n == 1:
```
Comment (PythonCHB) [inline students/GKim/lesson02/series.py:16]: did you try without this check? turns out you don't need it ... can you see why?

File: students/GKim/lesson03/list_lab.py [hunk 10]
```diff
@@ -0,0 +1,96 @@
+
```
Comment (PythonCHB) [inline students/GKim/lesson03/list_lab.py:1]: now you are getting ahead of yourself -- what will you do in class?

**Review by PythonCHB — APPROVED** [review]
nice work!

I haven't reviewed the list lab and slicing yet -- be sure to ask for a review of those if you want them when you do your next PR.

**Pull request merged**